runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -38,8 +39,42 @@ namespace Catalyst::Runtime {
 /**
  * @brief Global quantum device unique pointer.
  */
-thread_local static std::unique_ptr<ExecutionContext> CTX = nullptr;
+static std::unique_ptr<ExecutionContext> CTX = nullptr;
 
+/**
+ * @brief Thread local device pointer with internal linkage.
+ */
+thread_local static RTDevice *RTD_PTR = nullptr;
+
+/**
+ * @brief Initialize the device instance and update the value of RTD_PTR
+ * to the new initialized device pointer.
+ */
+[[nodiscard]] bool getOrCreateDevice(ExecutionContext *ec, std::string_view rtd_lib,
```

Comment:
I asked this in previous comment as well, but I don't quite understand why the EC would be passed in as a variable if it is global (a singleton essentially)?

Reply:
The return type (bool) and the function name (indicating a device might be returned) don't really match either.

Reply:
I'm also curious about the the `friend` attribute in the EC class, what purpose does it serve?

Reply:
Oops, I thought I got rid of all these friends methods after removing the `thread_local` device ID.
- Passing the EC as a variable was part of the `friend` method requirement (removed in the updated version)
- Returning a boolean is to ensue that we successfully initialize `RTD_PTR`. 
- I was using `friend` methods to access the device pool and manipulate devices using private methods and variables of the EC class. (removed in the updated version)

Thanks for your questions. They helped me properly tidy up the support.